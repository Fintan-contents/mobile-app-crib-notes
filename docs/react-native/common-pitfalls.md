---
title: ハマりがちな落とし穴
---

React Nativeで開発する際に起きた困り事や不具合についてまとめています。
