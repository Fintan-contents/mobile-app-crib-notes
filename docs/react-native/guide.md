---
title: 何らかのガイドを書きたい気持ちはある
---
