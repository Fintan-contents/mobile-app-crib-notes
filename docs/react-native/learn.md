---
title: About
hide_table_of_contents: true
---

このコンテンツはReact Nativeでのモバイルアプリケーション開発を学習するためのコンテンツです。

ReactでWebアプリケーションを実装できるがReact Nativeでモバイルアプリケーションを実装したことがないアプリケーション開発者を対象として、次のことについて学ぶことを目的としています。

- ReactとReact Nativeの違い
- React Nativeで作成したモバイルアプリケーションの動作確認方法

また、作成したモバイルアプリケーションの配布方法やテスト手法についてはこのコンテンツでは触れません。

<!-- textlint-disable ja-technical-writing/sentence-length,ja-technical-writing/max-comma,ja-spacing/ja-no-space-around-parentheses,jtf-style/3.3.かっこ類と隣接する文字の間のスペースの有無,ja-technical-writing/ja-no-mixed-period,ja-technical-writing/no-unmatched-pair -->

import {PageList} from '../../src/components';
const overviews = [
  {
    title: 'Getting Started',
    to: '/react-native/learn/getting-started',
    summary: <>React Nativeでのアプリ開発に必要な環境を用意します。</>
  },
  {
    title: 'アプリの実装前に',
    to: '/react-native/learn/basic-concepts',
    summary: <>React Nativeでアプリを実装し始める前に最低限必要となる前提知識や、React NativeとReact Navigationの基本的な知識について学びます。</>
  },
  {
    title: 'ToDoアプリの実装',
    to: '/react-native/learn/todo-app',
    summary: <>React Nativeで簡単なToDoアプリを実装します。</>
  },
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length,ja-technical-writing/max-comma,ja-spacing/ja-no-space-around-parentheses,jtf-style/3.3.かっこ類と隣接する文字の間のスペースの有無,ja-technical-writing/ja-no-mixed-period,ja-technical-writing/no-unmatched-pair -->
