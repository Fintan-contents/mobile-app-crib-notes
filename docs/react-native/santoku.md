---
title: Example Application
sidebar_label: About
hide_table_of_contents: true
---

React Nativeサンプルアプリケーションの設計ドキュメントです。

<!-- textlint-disable ja-technical-writing/sentence-length -->

import {PageList} from '../../src/components';
const overviews = [
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length -->
