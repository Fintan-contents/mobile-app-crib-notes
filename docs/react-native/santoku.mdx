---
title: Example Application
sidebar_label: About
hide_table_of_contents: true
---

<!-- textlint-disable ja-technical-writing/sentence-length -->

import {PageList} from '../../src/components';
const overviews = [
  {
    title: 'Development',
    summary: '開発環境やビルドに関するドキュメント',
    to: '/react-native/santoku/development',
  }
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length -->
