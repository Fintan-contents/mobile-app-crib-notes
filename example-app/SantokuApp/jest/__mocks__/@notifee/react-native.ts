import notifee from '@notifee/react-native';
export * from '@notifee/react-native/dist/types/Library';
export * from '@notifee/react-native/dist/types/Notification';
export * from '@notifee/react-native/dist/types/Trigger';
export * from '@notifee/react-native/dist/types/NotificationIOS';
export * from '@notifee/react-native/dist/types/NotificationAndroid';

const mock: typeof notifee = {
  cancelAllNotifications: jest.fn(),
  cancelDisplayedNotifications: jest.fn(),
  cancelTriggerNotifications: jest.fn(),
  cancelNotification: jest.fn(),
  cancelDisplayedNotification: jest.fn(),
  cancelTriggerNotification: jest.fn(),
  createChannel: jest.fn(),
  createChannels: jest.fn(),
  createChannelGroup: jest.fn(),
  createChannelGroups: jest.fn(),
  deleteChannel: jest.fn(),
  deleteChannelGroup: jest.fn(),
  displayNotification: jest.fn(),
  openAlarmPermissionSettings: jest.fn(),
  createTriggerNotification: jest.fn(),
  getTriggerNotificationIds: jest.fn(),
  getDisplayedNotifications: jest.fn(),
  getTriggerNotifications: jest.fn(),
  getChannel: jest.fn(),
  isChannelCreated: jest.fn(),
  isChannelBlocked: jest.fn(),
  getChannels: jest.fn(),
  getChannelGroup: jest.fn(),
  getChannelGroups: jest.fn(),
  getInitialNotification: jest.fn(),
  onBackgroundEvent: jest.fn(),
  onForegroundEvent: jest.fn(),
  openNotificationSettings: jest.fn(),
  registerForegroundService: jest.fn(),
  stopForegroundService: jest.fn(),
  requestPermission: jest.fn(),
  setNotificationCategories: jest.fn(),
  getNotificationCategories: jest.fn(),
  getNotificationSettings: jest.fn(),
  getBadgeCount: jest.fn(),
  setBadgeCount: jest.fn(),
  incrementBadgeCount: jest.fn(),
  decrementBadgeCount: jest.fn(),
  openBatteryOptimizationSettings: jest.fn(),
  isBatteryOptimizationEnabled: jest.fn(),
  getPowerManagerInfo: jest.fn(),
  openPowerManagerSettings: jest.fn(),
  hideNotificationDrawer: jest.fn(),
  SDK_VERSION: '5.4.1',
};

// 複数のファイルでnotifeeをimportしていた場合に、redefineされないようにモックが存在するか確認
if (!__mocks.notifee) {
  Object.defineProperty(__mocks, 'notifee', {value: mock});
}

// テストケースごとにモックは初期化しておく。
beforeEach(() =>
  Object.values(mock).forEach(fn => {
    if (jest.isMockFunction(fn)) {
      fn.mockClear();
    }
  }),
);

export default mock;
