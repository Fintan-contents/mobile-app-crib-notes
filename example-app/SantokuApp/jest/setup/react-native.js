// https://reactnavigation.org/docs/testing#mocking-native-modules
// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

// !!! NOTICE !!!
// The React Navigation documentation says that `react-native-reanimated` and `react-native-gesture-handler`
// also need to be configured with mocks.
// But since React Navigation v6 does not already depend on those packages, mocking those packages is not necessary
// for React Navigation v6.
//
// If you want to use `react-native-reanimated` and `react-native-gesture-handler`, add mock according to the guide
// provided by each package.
// - `react-native-reanimated`: https://docs.swmansion.com/react-native-reanimated/docs/guide/testing/
// - `react-native-gesture-handler`: https://docs.swmansion.com/react-native-gesture-handler/docs/guides/testing
