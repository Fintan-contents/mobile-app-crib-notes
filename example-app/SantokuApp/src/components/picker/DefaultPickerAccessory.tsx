import React from 'react';
import {StyleSheet, TextProps, TouchableOpacityProps, View, ViewProps} from 'react-native';

import {PickerAccessoryButton, PickerAccessoryButtonProps} from './PickerAccessoryButton';

export type DefaultPickerAccessoryProps = {
  containerProps?: ViewProps;
  onDelete?: () => void;
  deleteLabel?: string;
  deleteTextProps?: TextProps;
  deleteTextComponent?: PickerAccessoryButtonProps['textComponent'];
  deleteTouchableContainerProps?: TouchableOpacityProps;
  onCancel?: () => void;
  cancelLabel?: string;
  cancelTextProps?: TextProps;
  cancelTextComponent?: PickerAccessoryButtonProps['textComponent'];
  cancelTouchableContainerProps?: TouchableOpacityProps;
  onDone?: () => void;
  doneLabel?: string;
  doneTextProps?: TextProps;
  doneTextComponent?: PickerAccessoryButtonProps['textComponent'];
  doneTouchableContainerProps?: TouchableOpacityProps;
};

export const DefaultPickerAccessory: React.FC<DefaultPickerAccessoryProps> = ({
  containerProps: {style, ...containerProps} = {},
  onDelete,
  deleteTextProps,
  deleteLabel,
  deleteTextComponent,
  deleteTouchableContainerProps,
  onCancel,
  cancelTextProps,
  cancelLabel,
  cancelTextComponent,
  cancelTouchableContainerProps,
  onDone,
  doneTextProps,
  doneLabel,
  doneTextComponent,
  doneTouchableContainerProps,
}) => {
  return (
    <View style={StyleSheet.flatten([styles.container, style])} {...containerProps}>
      {deleteLabel && (
        <>
          <PickerAccessoryButton
            onPress={onDelete}
            defaultStyle={styles.destructiveTextStyle}
            textProps={deleteTextProps}
            textComponent={deleteTextComponent}
            touchableContainerProps={deleteTouchableContainerProps}
            label={deleteLabel}
          />
          <View style={styles.flex} />
        </>
      )}
      {cancelLabel && (
        <PickerAccessoryButton
          onPress={onCancel}
          defaultStyle={styles.cancelTextStyle}
          textProps={cancelTextProps}
          textComponent={cancelTextComponent}
          touchableContainerProps={cancelTouchableContainerProps}
          label={cancelLabel}
        />
      )}
      {doneLabel && (
        <PickerAccessoryButton
          onPress={onDone}
          defaultStyle={styles.okTextStyle}
          textProps={doneTextProps}
          textComponent={doneTextComponent}
          touchableContainerProps={doneTouchableContainerProps}
          label={doneLabel}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    paddingHorizontal: 10,
    paddingVertical: 8,
  },
  destructiveTextStyle: {
    color: '#d9545e',
    fontWeight: 'bold',
    paddingHorizontal: 10,
  },
  cancelTextStyle: {
    color: '#4577CC',
    fontWeight: 'bold',
    paddingHorizontal: 10,
  },
  okTextStyle: {
    color: '#4577CC',
    fontWeight: 'bold',
    paddingHorizontal: 10,
  },
  flex: {
    flex: 1,
  },
});
