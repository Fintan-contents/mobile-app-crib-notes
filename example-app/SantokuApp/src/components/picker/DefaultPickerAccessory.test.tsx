import {fireEvent, render, screen} from '@testing-library/react-native';
import React from 'react';
import {ViewProps} from 'react-native';
import {Text, TextProps} from 'react-native-elements';

import {DefaultPickerAccessory} from './DefaultPickerAccessory';

describe('PickerAccessory with `xxxLabel`', () => {
  it('renders successfully only with delete link', () => {
    const onDelete = jest.fn();
    render(
      <DefaultPickerAccessory
        deleteLabel="delete"
        onDelete={onDelete}
        deleteTouchableContainerProps={{testID: 'deleteLink'}}
      />,
    );
    const deleteLink = screen.getByTestId('deleteLink');
    const cancelLink = screen.queryByTestId('cancelLink');
    const doneLink = screen.queryByTestId('doneLink');
    const deleteText = screen.queryByText('delete');
    fireEvent.press(deleteLink);
    expect(onDelete).toHaveBeenCalledTimes(1);
    expect(deleteText).not.toBeNull();
    expect(cancelLink).toBeNull();
    expect(doneLink).toBeNull();
  });

  it('renders successfully only with cancel link', () => {
    const onCancel = jest.fn();
    render(
      <DefaultPickerAccessory
        cancelLabel="cancel"
        onCancel={onCancel}
        cancelTouchableContainerProps={{testID: 'cancelLink'}}
      />,
    );
    const deleteLink = screen.queryByTestId('deleteLink');
    const cancelLink = screen.getByTestId('cancelLink');
    const doneLink = screen.queryByTestId('doneLink');
    const cancelText = screen.queryByText('cancel');
    fireEvent.press(cancelLink);
    expect(onCancel).toHaveBeenCalledTimes(1);
    expect(cancelText).not.toBeNull();
    expect(deleteLink).toBeNull();
    expect(doneLink).toBeNull();
  });

  it('renders successfully only with done link', () => {
    const onDone = jest.fn();
    render(
      <DefaultPickerAccessory doneLabel="done" onDone={onDone} doneTouchableContainerProps={{testID: 'doneLink'}} />,
    );
    const deleteLink = screen.queryByTestId('deleteLink');
    const cancelLink = screen.queryByTestId('cancelLink');
    const doneLink = screen.getByTestId('doneLink');
    const doneText = screen.queryByText('done');
    fireEvent.press(doneLink);
    expect(onDone).toHaveBeenCalledTimes(1);
    expect(doneText).not.toBeNull();
    expect(deleteLink).toBeNull();
    expect(cancelLink).toBeNull();
  });
});

describe('PickerAccessory with all props', () => {
  it('should be applied properly with default text component', () => {
    const onDelete = jest.fn();
    const onCancel = jest.fn();
    const onDone = jest.fn();
    render(
      <DefaultPickerAccessory
        containerProps={{style: {backgroundColor: 'red'}, testID: 'PickerAccessory'}}
        onDelete={onDelete}
        onCancel={onCancel}
        onDone={onDone}
        deleteLabel="delete"
        cancelLabel="cancel"
        doneLabel="done"
        deleteTextProps={{style: {color: 'yellow'}, testID: 'deleteText'}}
        cancelTextProps={{style: {color: 'green'}, testID: 'cancelText'}}
        doneTextProps={{style: {color: 'aqua'}, testID: 'doneText'}}
        deleteTouchableContainerProps={{testID: 'deleteLink'}}
        cancelTouchableContainerProps={{testID: 'cancelLink'}}
        doneTouchableContainerProps={{testID: 'doneLink'}}
      />,
    );
    expect(screen).toMatchSnapshot('PickerAccessory with all props(default text component ).');
    const pickerAccessory = screen.getByTestId('PickerAccessory');
    const deleteLink = screen.getByTestId('deleteLink');
    const cancelLink = screen.getByTestId('cancelLink');
    const doneLink = screen.getByTestId('doneLink');
    const deleteTextComponent = screen.getByTestId('deleteText');
    const cancelTextComponent = screen.getByTestId('cancelText');
    const doneTextComponent = screen.getByTestId('doneText');
    const deleteText = screen.queryByText('delete');
    const cancelText = screen.queryByText('cancel');
    const doneText = screen.queryByText('done');

    fireEvent.press(deleteLink);
    fireEvent.press(cancelLink);
    fireEvent.press(doneLink);

    const pickerAccessoryProps = pickerAccessory.props as ViewProps;
    const deleteTextProps = deleteTextComponent.props as TextProps;
    const cancelTextProps = cancelTextComponent.props as TextProps;
    const doneTextProps = doneTextComponent.props as TextProps;
    expect(pickerAccessoryProps.style).toEqual({
      flexDirection: 'row',
      justifyContent: 'flex-end',
      paddingHorizontal: 10,
      paddingVertical: 8,
      backgroundColor: 'red',
    });
    expect(onDelete).toHaveBeenCalledTimes(1);
    expect(onCancel).toHaveBeenCalledTimes(1);
    expect(onDone).toHaveBeenCalledTimes(1);
    expect(deleteTextProps.style).toEqual({color: 'yellow', fontWeight: 'bold', paddingHorizontal: 10});
    expect(cancelTextProps.style).toEqual({color: 'green', fontWeight: 'bold', paddingHorizontal: 10});
    expect(doneTextProps.style).toEqual({color: 'aqua', fontWeight: 'bold', paddingHorizontal: 10});
    expect(deleteText).not.toBeNull();
    expect(cancelText).not.toBeNull();
    expect(doneText).not.toBeNull();
  });

  it('should be applied properly with custom text component', () => {
    const onDelete = jest.fn();
    const onCancel = jest.fn();
    const onDone = jest.fn();
    const DeleteTextComponent = (
      <Text style={{color: 'grey', backgroundColor: 'yellow'}} testID="deleteText">
        deleteWithRNEText
      </Text>
    );
    const CancelTextComponent = (
      <Text style={{color: 'black', backgroundColor: 'green'}} testID="cancelText">
        cancelWithRNEText
      </Text>
    );
    const DoneTextComponent = (
      <Text style={{color: 'blue', backgroundColor: 'aqua'}} testID="doneText">
        doneWithRNEText
      </Text>
    );
    render(
      <DefaultPickerAccessory
        containerProps={{style: {backgroundColor: 'red'}, testID: 'PickerAccessory'}}
        onDelete={onDelete}
        onCancel={onCancel}
        onDone={onDone}
        deleteLabel="delete"
        cancelLabel="cancel"
        doneLabel="done"
        deleteTextProps={{style: {color: 'yellow'}, testID: 'deleteText'}}
        cancelTextProps={{style: {color: 'green'}, testID: 'cancelText'}}
        doneTextProps={{style: {color: 'aqua'}, testID: 'doneText'}}
        deleteTouchableContainerProps={{testID: 'deleteLink'}}
        cancelTouchableContainerProps={{testID: 'cancelLink'}}
        doneTouchableContainerProps={{testID: 'doneLink'}}
        deleteTextComponent={DeleteTextComponent}
        cancelTextComponent={CancelTextComponent}
        doneTextComponent={DoneTextComponent}
      />,
    );
    expect(screen).toMatchSnapshot('PickerAccessory with all props(custom text component ).');
    const pickerAccessory = screen.getByTestId('PickerAccessory');
    const deleteLink = screen.getByTestId('deleteLink');
    const cancelLink = screen.getByTestId('cancelLink');
    const doneLink = screen.getByTestId('doneLink');
    const deleteTextComponent = screen.getByTestId('deleteText');
    const cancelTextComponent = screen.getByTestId('cancelText');
    const doneTextComponent = screen.getByTestId('doneText');
    const deleteText = screen.queryByText('deleteWithRNEText');
    const cancelText = screen.queryByText('cancelWithRNEText');
    const doneText = screen.queryByText('doneWithRNEText');

    fireEvent.press(deleteLink);
    fireEvent.press(cancelLink);
    fireEvent.press(doneLink);

    const pickerAccessoryProps = pickerAccessory.props as ViewProps;
    const deleteTextProps = deleteTextComponent.props as TextProps;
    const cancelTextProps = cancelTextComponent.props as TextProps;
    const doneTextProps = doneTextComponent.props as TextProps;
    expect(pickerAccessoryProps.style).toEqual({
      flexDirection: 'row',
      justifyContent: 'flex-end',
      paddingHorizontal: 10,
      paddingVertical: 8,
      backgroundColor: 'red',
    });
    expect(onDelete).toHaveBeenCalledTimes(1);
    expect(onCancel).toHaveBeenCalledTimes(1);
    expect(onDone).toHaveBeenCalledTimes(1);
    expect(deleteTextProps.style).toEqual({color: 'grey', backgroundColor: 'yellow'});
    expect(cancelTextProps.style).toEqual({color: 'black', backgroundColor: 'green'});
    expect(doneTextProps.style).toEqual({color: 'blue', backgroundColor: 'aqua'});
    expect(deleteText).not.toBeNull();
    expect(cancelText).not.toBeNull();
    expect(doneText).not.toBeNull();
  });
});
