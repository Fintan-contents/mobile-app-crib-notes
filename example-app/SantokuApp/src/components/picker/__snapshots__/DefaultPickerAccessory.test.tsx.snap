// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerAccessory with all props should be applied properly with custom text component: PickerAccessory with all props(custom text component ). 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "paddingHorizontal": 10,
      "paddingVertical": 8,
    }
  }
  testID="PickerAccessory"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="deleteLink"
  >
    <Text
      style={
        Object {
          "backgroundColor": "yellow",
          "color": "grey",
        }
      }
      testID="deleteText"
    >
      deleteWithRNEText
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="cancelLink"
  >
    <Text
      style={
        Object {
          "backgroundColor": "green",
          "color": "black",
        }
      }
      testID="cancelText"
    >
      cancelWithRNEText
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="doneLink"
  >
    <Text
      style={
        Object {
          "backgroundColor": "aqua",
          "color": "blue",
        }
      }
      testID="doneText"
    >
      doneWithRNEText
    </Text>
  </View>
</View>
`;

exports[`PickerAccessory with all props should be applied properly with default text component: PickerAccessory with all props(default text component ). 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "paddingHorizontal": 10,
      "paddingVertical": 8,
    }
  }
  testID="PickerAccessory"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="deleteLink"
  >
    <Text
      style={
        Object {
          "color": "yellow",
          "fontWeight": "bold",
          "paddingHorizontal": 10,
        }
      }
      testID="deleteText"
    >
      delete
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="cancelLink"
  >
    <Text
      style={
        Object {
          "color": "green",
          "fontWeight": "bold",
          "paddingHorizontal": 10,
        }
      }
      testID="cancelText"
    >
      cancel
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="doneLink"
  >
    <Text
      style={
        Object {
          "color": "aqua",
          "fontWeight": "bold",
          "paddingHorizontal": 10,
        }
      }
      testID="doneText"
    >
      done
    </Text>
  </View>
</View>
`;
